### DISCUSSION TITLE
[24/10/2024]: Community Call

### DISCUSSION THREAD

**Discussion by sujaya-sys** [IMGARENA/checkly-operator#48] [Community Meetings]
Call with Alen, platform engineer, who provided some valuable feedback on the project:

**Documentation**
- Conduct general clean up (e.g. ensure consistent capitalization)
- Rethink structure: The readme starts by highlighting "development" topics, we should instead move "usage" focused content to the top

**feature requests** (Added notes to the corresponding issue)

- Allow installation via Helm: https://github.com/checkly/checkly-operator/issues/42#issuecomment-2435018433
- Add support for Checkly Agent: https://github.com/checkly/checkly-operator/issues/43#issuecomment-2435026200
- Auto provisioning via Ingress: https://github.com/checkly/checkly-operator/discussions/46#discussioncomment-11027331
- API Checks: Keep it lean - the suggestion is to not allow the creation of multiple endpoint checks within one API check configuration to avoid bloated config files

**Security**
- Teams often work with tools that enforce cluster- and namespace scoped policies e.g. to determine which users/roles can manage certain objects
- We might want to look into the operator taking these kind of security settings / policies into account i.e. only allow users with certain permissions to create / modify checks for example

**Observability**
- We should emit prometheus metrics from the operator / agent pods
- It'll also be great to allow for the creation of standard resources (such as pod / service monitor components) that can be automatically be picked up by prometheus
- Configuration of prometheus rules would be equally helpful i.e. define what you want to be alerted on

**Comment by akosveres** [comment]
>We should emit prometheus metrics from the operator / agent pods

We currently do this for the operator pods, there are default metrics from the kubebuilder project: https://github.com/checkly/checkly-operator/blob/main/cmd/main.go#L60 - We should emit extra metrics which are related to the calls we make to the checklyhq.com APIs.
